src/plugins/es_ui_shared/__packages_do_not_import__/errors/handle_es_error.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ApiError } from '@elastic/elasticsearch';
+import { IKibanaResponse, KibanaResponseFactory } from 'kibana/server';
+
+interface EsErrorHandlerParams {
+  error: ApiError;
+  response: KibanaResponseFactory;
+}
+export const handleEsError = ({ error, response }: EsErrorHandlerParams): IKibanaResponse => {
+  // error.name is slightly better in terms of performance, since all errors now have name property
+  if (error.name === 'ResponseError') {
+    return response.customError({
+      // we can ignore typescript error, since error is a ResponseError
```

Comment:
I think here it needs to be `ApiResponse` since es client can return different types or errors (`TimeoutError`, `ConfigurationError` etc), but they all will have a name property.

Reply:
Ah, right. I read the comment and forgot it was scoped within the `if` block 😅.

What about something like this?

```
    const { statusCode, body } = error as ResponseError;

    return response.customError({
      statusCode,
      body: { message: body.error?.reason },
    });
```

Reply:
thank you @alisonelizabeth , I totally forgot type casting exists 🤦🏻‍♀️